src/core/graph/graph.ts

```
@@ -354,4 +357,139 @@ export class TransitGraph {
+  calculateLineConnectivity(): void {
+    const lineIds = Object.keys(this.lines);
+    console.log(`[Graph] Calculating line connectivity for ${lineIds.length} lines`);
+    
+    // Initialize the line connectivity matrix with Infinity values
+    lineIds.forEach(fromLineId => {
+      this.lineConnectivityMatrix[fromLineId] = {};
+      lineIds.forEach(toLineId => {
+        // Set to 0 for same line, Infinity for different lines
+        this.lineConnectivityMatrix[fromLineId][toLineId] = 
+          fromLineId === toLineId ? 0 : Infinity;
+      });
```

Biome (1.9.4)

[error] Use Number.POSITIVE_INFINITY instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.POSITIVE_INFINITY instead.


(lint/style/useNumberNamespace)